### PR TITLE
SDP-465: Update PATCH {AP}/transaction body with the appropriate fields

### PR DESCRIPTION
**Pull request by marcelosalloum**
### What

Update PATCH {AP}/transaction body with the appropriate fields.

### Why

We had a task to review if we were PATCHING the correct fields, and it turns out we were sending no-op fields that were being ignored by the AP.

#### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).
* [x] This PR's title starts with the name of the package, area, or subject affected by the change.

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [x] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/anchorplatform/object_schemas.go [hunk 1]
```diff
@@ -78,39 +91,46 @@ type APSep24Transaction struct {
...
+// type APSep24TransactionPatchPostError struct {
 // 	ID                  string                 `json:"id"`
 // 	SEP                 string                 `json:"sep,omitempty"`
-// 	Status              APTransactionStatus    `json:"status,omitempty"` // Error
 // 	StellarTransactions []APStellarTransaction `json:"stellar_transactions,omitempty"`
 // 	Message             string                 `json:"message,omitempty"` // Error message
+// 	Status              APTransactionStatus    `json:"status,omitempty"`  // Error
+// 	// TODO: update the AP version when source_account becomes patchable
+// 	SourceAccount string `json:"source_account,omitempty"`
 // }
```
Comment (marcelosalloum) [inline internal/anchorplatform/object_schemas.go:134]: These objects were created to make it simple for us to know which fields we will want to update on case of Registration, PaymentSuccess of PaymentFailure.

File: internal/serve/httphandler/verifiy_receiver_registration_handler.go [hunk 2]
```diff
@@ -194,14 +194,10 @@ func (v VerifyReceiverRegistrationHandler) processReceiverWalletOTP(ctx context.
...
 // processAnchorPlatformID PATCHes the transaction on the AnchorPlatform with the status "pending_anchor" and updates the local database accordingly.
 func (v VerifyReceiverRegistrationHandler) processAnchorPlatformID(ctx context.Context, sep24Claims *anchorplatform.SEP24JWTClaims) error {
-	apTxPatch := anchorplatform.APSep24Transaction{
-		ID:                 sep24Claims.TransactionID(),
-		SEP:                "24",
-		Status:             anchorplatform.APTransactionStatusPendingAnchor,
-		Kind:               "deposit",
-		DestinationAccount: sep24Claims.SEP10StellarAccount(),
-		Memo:               sep24Claims.SEP10StellarMemo(),
-		KYCVerified:        true,
```
Comment (marcelosalloum) [inline internal/serve/httphandler/verifiy_receiver_registration_handler.go:204]: These fields were being sent for no reason. The AP API does not allow them to be patched.

File: internal/anchorplatform/platform_api_service_mock.go [hunk 3]
```diff
@@ -10,8 +10,12 @@ type AnchorPlatformAPIServiceMock struct {
...
 }
 
-func (a *AnchorPlatformAPIServiceMock) PatchAnchorTransactionsPostRegistration(ctx context.Context, apTxPatch ...APSep24Transaction) error {
-	args := a.Called(ctx, apTxPatch)
+func (a *AnchorPlatformAPIServiceMock) PatchAnchorTransactionsPostRegistration(ctx context.Context, apTxPatch ...APSep24TransactionPatchPostRegistration) error {
+	inputArgs := []interface{}{ctx}
+	for _, patch := range apTxPatch {
+		inputArgs = append(inputArgs, patch)
+	}
+	args := a.Called(inputArgs...)
```
Comment (marcelosalloum) [inline internal/anchorplatform/platform_api_service_mock.go:18]: This change was made to support variadic arguments in the Mock.

**Review by marcelosalloum — COMMENTED** [review]
Adding some clarification comments below.

File: internal/serve/httphandler/verifiy_receiver_registration_handler_test.go [hunk 4]
```diff
@@ -693,20 +689,16 @@ func Test_VerifyReceiverRegistrationHandler_VerifyReceiverRegistration(t *testin
...
-		apTxPatches := []anchorplatform.APSep24Transaction{{
-			ID:                 "test-transaction-id",
-			Status:             "pending_anchor",
-			SEP:                "24",
-			Kind:               "deposit",
-			DestinationAccount: validClaims.SEP10StellarAccount(),
-			Memo:               validClaims.SEP10StellarMemo(),
-			KYCVerified:        true,
-		}}
+		apTxPatche := anchorplatform.APSep24TransactionPatchPostRegistration{
```
Comment (CaioTeixeira95) [inline internal/serve/httphandler/verifiy_receiver_registration_handler_test.go:692]: Perhaps `apTxPatch`?

Comment (marcelosalloum) [reply internal/serve/httphandler/verifiy_receiver_registration_handler_test.go]: Addressed in https://github.com/stellar/stellar-disbursement-platform-backend/pull/40/commits/f6a244b2655f8926423b33cfc0d0efaefba13996

**Review by CaioTeixeira95 — APPROVED** [review]
LGTM

**Pull request merged**